prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -58,10 +78,181 @@ class Hl7Serializer(val metadata: Metadata) {
         outputStream.write(createFooters(report).toByteArray())
     }
 
+    /*
+     * Read in a file
+     */
+    fun convertBatchMessagesToMap(message: String, schema: Schema): Hl7Mapping {
+        val mappedRows: MutableMap<String, MutableList<String>> = mutableMapOf()
+        val errors = mutableListOf<String>()
+        val warnings = mutableListOf<String>()
+        val rowResults = mutableListOf<RowResult>()
+        val reg = "(\r|\n)".toRegex()
+        val cleanedMessage = reg.replace(message, "\r")
+        val messageLines = cleanedMessage.split("\r")
+        val nextMessage = StringBuilder()
+
+        fun deconstructStringMessage() {
+            println("Examining:\n$nextMessage")
+            val parsedMessage = convertMessageToMap(nextMessage.toString(), schema)
+            errors.addAll(parsedMessage.errors)
+            warnings.addAll(parsedMessage.warnings)
+            rowResults.add(parsedMessage)
+            nextMessage.clear()
+            parsedMessage.row.forEach { (k, v) ->
+                if (!mappedRows.containsKey(k))
+                    mappedRows[k] = mutableListOf()
+
+                mappedRows[k]?.addAll(v)
+            }
+        }
+
+        messageLines.forEach {
+            if (it.startsWith("FHS"))
+                return@forEach
+            if (it.startsWith("BHS"))
+                return@forEach
+            if (it.startsWith("BTS"))
+                return@forEach
+            if (it.startsWith("FTS"))
+                return@forEach
+
+            if (nextMessage.isNotEmpty() && it.startsWith("MSH")) {
+                deconstructStringMessage()
+            }
+            nextMessage.append("$it\r")
+        }
+
+        // catch the last message
+        if (nextMessage.isNotEmpty()) {
+            deconstructStringMessage()
+        }
+
+        return Hl7Mapping(mappedRows, rowResults, errors, warnings)
+    }
+
+    fun convertMessageToMap(message: String, schema: Schema): Hl7Serializer.RowResult {
+        // safely merge into the set. might not be necessary
+        fun mergeIntoMappedRows(mappedRows: MutableMap<String, MutableSet<String>>, key: String, value: String) {
+            if (!mappedRows.containsKey(key)) error("Map doesn't contain key $key")
+            // get the existing values
+            val existingValues = mappedRows[key] ?: emptySet()
+            // if the existing value doesn't exist, add it in
+            if (!existingValues.contains(value)) {
+                mappedRows[key]?.add(value)
+            }
```

Comment:
This might not be necessary, but originally it wasn't a `Set` but a `List` so I wanted to not add duplicate values if we were using `hl7OutputFields`. We can probably remove and let `Set` do all the work.